src/main/java/org/jabref/gui/entryeditor/EntryEditorPreferences.java

```diff
@@ -24,7 +24,17 @@
     private boolean avoidOverwritingCiteKey;
     private final boolean shouldShowLatexCitationsTab;
 
-    public EntryEditorPreferences(Map<String, Set<Field>> entryEditorTabList, FieldWriterPreferences fieldWriterPreferences, ImportFormatPreferences importFormatPreferences, List<Field> customTabFieldNames, boolean shouldShowRecommendationsTab, boolean isMrdlibAccepted, boolean shouldShowLatexCitationsTab, boolean showSourceTabByDefault, BibtexKeyPatternPreferences bibtexKeyPatternPreferences, KeyBindingRepository keyBindings, boolean avoidOverwritingCiteKey) {
+    public EntryEditorPreferences(Map<String, Set<Field>> entryEditorTabList,
+                                  FieldWriterPreferences fieldWriterPreferences,
+                                  ImportFormatPreferences importFormatPreferences,
+                                  List<Field> customTabFieldNames,
+                                  boolean shouldShowRecommendationsTab,
```

Comment:
This is probably a huge thing, but what about replacing this many boolean parameters with an enum?

Reply:
I think I'll leave that to my successor. 😉 I really wanted to focus on extracting the single calls to JabRefPreferences::get and put out of the codebase.
As soon as this is done, it should be easier to distinguish between preferences model and logic.
But we should keep this idea in mind.